### ISSUE TITLE
Can't run more than one Cuda task

### ISSUE THREAD

**Issue by sparkasaurusRex** [StanfordLegion/legion#497]
I'm launching several leaf tasks with `__demand(__cuda)` and I get this error.
```
terra: /home/users/ellish/repos/legion/runtime/legion/legion_domain.inl:1015: Legion::Domain::operator Legion::Rect<DIM, T>() const [with int DIM = 1; T = long long int]: Assertion `is_id == 0' failed.
*** Caught a fatal signal: SIGABRT(6) on node 0/1
[0 - 7fbc4b657780] {4}{threads}: reservation ('DMA request queue') cannot be satisfied
```

It actually worked when I had at most one `__demand(__cuda)`. Earlier, I was having a similar problem to issue #483 which was fixed by commit 284a71b.

**Comment by magnatelee** [comment]
Launching multiple CUDA tasks alone doesn't give you this error. I suspect at least one of your CUDA task has a for loop over a sparse region. (I haven't added a code path for this.) Could you confirm this is the case? It would be helpful if you shared a small reproducer. BTW, for loops over sparse regions do not run efficiently on GPUs, so you should either make them dense or reduce the sparseness as much as you can.

**Comment by sparkasaurusRex** [comment]
I believe this is the case, although I don’t think my data is very sparse. What exactly is the definition of a sparse region? I’ll try to put together a smaller reproducer.

**Comment by magnatelee** [comment]
A region is sparse when the points in its index space are not contiguous. I don't think a reproducer is necessary anymore, as I have a clear idea what goes wrong. I'll try to push a fix soon.

**Comment by magnatelee** [comment]
This should be fixed by cf1e1825. @sparkasaurusRex Could you pull and try again?

**Comment by sparkasaurusRex** [comment]
It looks like this issue was fixed, thanks!
